### PR TITLE
Cherry-pick Babelfish-related commits from BABEL_1_X_DEV__PG_13_6 - part 2

### PR DESCRIPTION
**Pull request by 2jungkook**
# WARNING: DO NOT SQUASH COMMIT THIS PR!

### Description

Cherry-picked 32 Babelfish-related commits from `BABEL_1_X_DEV__PG_13_6`. During the cherry-pick, there were 5 merge conflicts from the below commits.

```
21b413a873 Support CHECKSUM(*) and CHECKSUM(c1, c2, ...)
a7c52140dd Part 1: Support inserted transition table for INSTEAD OF INSERT
07dedf85fc Support sys.dm_exec_sessions and sys.dm_exec_connections dynamic management view
e2df156806 Fix misevaluation of STABLE parameters in CALL within plpgsql.
3a9ac9ad19 Do not filter pushes based on branch name
```

Conflicted files are
```
.github/workflows/main.yml
src/backend/commands/functioncmds.c
src/backend/postmaster/pgstat.c
src/backend/utils/adt/pgstatfuncs.c
src/backend/commands/trigger.c
src/backend/parser/parse_expr.c
```

Original merge conflict diff: [pg14merge_2_diff.txt](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/files/8339028/pg14merge_2_diff.txt)

### Check List

- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/backend/commands/functioncmds.c [hunk 1]
```diff
@@ -2447,7 +2447,11 @@ ExecuteCallStmt(CallStmt *stmt, ParamListInfo params, bool atomic, DestReceiver
 		i++;
 	}
 
-	/*
+	/* Get rid of temporary snapshot for arguments, if we made one */
```
Comment (nasbyj) [inline src/backend/commands/functioncmds.c:2450]: I don't think this is what we want here. I'm pretty sure the original community code had this line. If that's the case, the fact that this commit is adding it back in means that the community actually removed this bit in 14.x.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -2,11 +2,9 @@ name: CI
 
 on:
   push:
-    branches:
-      - 13_4
   pull_request:
     branches:
-      - 13_4
+      - BABEL_2_X_DEV__PG_14_2
```
Comment (nasbyj) [inline .github/workflows/main.yml]: Please remove this line; the point of the commit was to stop all filtering of the workflow.

**Review by nasbyj — CHANGES_REQUESTED** [review]

File: src/backend/commands/functioncmds.c (hunk 1 above)
Comment (2jungkook) [reply src/backend/commands/functioncmds.c:2450]: Community code has this line.
https://github.com/postgres/postgres/blob/REL_14_2/src/backend/commands/functioncmds.c

File: .github/workflows/main.yml (hunk 2 above)
Comment (2jungkook) [reply .github/workflows/main.yml]: okay, will update it.

**Review by forestkeeper — DISMISSED** [review]
LGTM

**Review by thephantomthief — DISMISSED** [review]

**Review by forestkeeper — APPROVED** [review]

**Review by thephantomthief — APPROVED** [review]

**Review by forestkeeper — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 2 above)
Comment (nasbyj) [reply .github/workflows/main.yml]: This line is still showing up. I think the correct thing to do is simply remove this commit entirely.

Comment (nasbyj) [reply .github/workflows/main.yml]: Wait, I take that back. We still need the commit (to remove the 13_4 bit). But the commit should not add `BABEL_2_X_DEV__PG_14_2`, unless I'm missing something?

Comment (nasbyj) [reply .github/workflows/main.yml]: I've verified this ultimately gets resolved by another commit at the end of this PR.

**Review by nasbyj — APPROVED** [review]

**Pull request merged**